std/src/main/scala/algebra/std/map.scala

```diff
@@ -130,3 +133,25 @@ class MapRng[K, V](implicit V: Rng[V]) extends MapSemiring[K, V] with Rng[Map[K,
   override def minus(x: Map[K, V], y: Map[K, V]): Map[K, V] =
     subtractMap(x, y)(V.minus)(V.negate)
 }
+
+class MapGenBool[K, V](implicit GBV: GenBool[V], EqV: Eq[V]) extends GenBool[Map[K, V]] {
```

Comment:
this is sparse, unlike the other implementations above. We could take `Eq[V]` on anything with zero on the values, and do something similar. We did this in algebird (we messed up a bit and didn't define an Eq or Equiv, so we just use `==`), and often the sparseness was an issues.

I think in general, have a method to remove zeros is a better approach and less surprising (and also consistent with other items).

What do you think of the above?